bpf/bpf_lxc.c

```diff
@@ -684,6 +724,32 @@ static inline int __inline__ ipv6_policy(struct __sk_buff *skb, int ifindex, __u
 		ret = ct_create6(&CT_MAP6, &tuple, skb, 1, &ct_state_new);
 		if (IS_ERR(ret))
 			return ret;
+
+		ct_state_reply.proxy_port = ct_state_new.proxy_port;
+	}
+
+	if (ct_state_reply.proxy_port && (ret == CT_NEW || ret == CT_ESTABLISHED)) {
+		//union v6addr host_ip = HOST_IP;
+		//union v6addr lxc_ip = LXC_IP;
+		//__be32 sum;
+
+		if (l4_modify_port(skb, l4_off, TCP_DPORT_OFF, &csum_off,
+				   ct_state_reply.proxy_port, dport) < 0)
+			return DROP_WRITE_ERROR;
+
+#if 0
+		if (ipv6_store_daddr(skb, host_ip.addr, ETH_HLEN) < 0)
+			return DROP_WRITE_ERROR;
+		/* FIXME: Calculate the checksum fix in user space */
+		sum = csum_diff(lxc_ip.addr, 16, host_ip.addr, 16, 0);
+		if (csum_l4_replace(skb, l4_off, &csum_off, 0, sum, BPF_F_PSEUDO_HDR) < 0)
+			return DROP_CSUM_L4;
+#endif
+		/* Mark packet with PACKET_HOST and pass to host */
+		if (skb_change_type(skb, 0) < 0)
+			return DROP_WRITE_ERROR;
+
```

Comment:
The kernel side is mostly done but the proxymap is IPv4 only at this point. Aggreed on refactoring.